File: Final Project/MediVault_ DotNet/MediVault/MediVault.Backend/Models/Hospital.cs
namespace MediVault.Backend.Models
{
    public class Hospital
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User? User { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public string ContactNumber { get; set; } = string.Empty;
        public bool IsActive { get; set; } = true;
        
        // Navigation for consultations created by this hospital
        public ICollection<Consultation> Consultations { get; set; } = new List<Consultation>();
        
        // Navigation for files uploaded by this hospital
        public ICollection<PatientFile> PatientFiles { get; set; } = new List<PatientFile>();
        
        // Navigation for doctors in this hospital
        public ICollection<Doctor> Doctors { get; set; } = new List<Doctor>();
    }
}

File: Final Project/Medivault Spring/MediVault-Spring/MediVault/src/main/java/com/medivault/entites/Hospital.java
package com.medivault.entites;

import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "hospitals", uniqueConstraints = {
		@UniqueConstraint(name = "uk_hospitals_user_id", columnNames = "user_id")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class Hospital {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@OneToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "user_id", nullable = false)
	private User user;
	
	@Column(nullable = false, length = 150)
	private String name;
	
	@Column(nullable = false, columnDefinition = "TEXT")
	private String address;
	
	@Column(length = 15)
	private String contactNumber;
	
	@Column(nullable = false)
	private boolean isActive = false;
	
	@OneToMany(mappedBy = "hospital", cascade = CascadeType.ALL, orphanRemoval = true)
	private List<Doctor> doctors;
	
	@OneToMany(mappedBy = "hospital", cascade = CascadeType.ALL, orphanRemoval = true)
	private List<Consultation> consultations;
	
	@OneToMany(mappedBy = "uploadedByHospital", cascade = CascadeType.ALL)
	private List<PatientFile> patientFiles;
	
}
